modules/apps/collaboration/item-selector/item-selector-api/src/main/java/com/liferay/item/selector/ItemSelectorCriterion.java

```diff
@@ -23,11 +23,11 @@
  * entity type and return the specified {@link ItemSelectorReturnType}.
  *
  * <p>
- * Implementations of this interface can specify {@link ItemSelectorReturnType}s
- * that include additional fine-grained information. The information should
- * ideally be specified using primitive types (or very simple types that can be
- * serialized using JSON) in the {@link ItemSelectorReturnType} constructor.
- * Note that an empty constructor is mandatory.
+ * Implementations of this interface can provide additional information to have
+ * a fine grained detail about which entities can be selected. This should be
+ * done ideally using primitive types in the constructor (or very simple types
+ * that can be JSON serialized) and it is mandatory to have an empty
+ * constructor.
```

Comment:
How does an implementation "provide additional information"? Do you create class fields and setters/getters?

And with respect to constructors, the impl must create a no-args constructor. If the criterion is providing additional information, do you recommend creating a constructor with params for that information?